neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,104 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+#[derive(Debug, Default)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+macro_rules! add_mut_helpers {
+    ($($method:ident($self:ident $(,$arg:ident:$type:ty)?) => $expr:expr),+) => {
+        $(
+            pub fn $method(mut $self$(, $arg: $type)?) -> Self {
+                $expr;
+                $self
+            }
+        )+
+    }
+}
+
+impl NodeCluster {
+    pub fn new<F: Fn(usize) -> String>(capacity: usize, gen_dirname: F) -> Self {
+        Self {
+            dirs: (0..capacity)
+                .map(|index| {
+                    tempfile::Builder::new().prefix(&gen_dirname(index)).tempdir().unwrap()
+                })
+                .collect(),
+            ..Default::default()
+        }
+    }
+
+    add_mut_helpers! {
+        set_heavy_test(self) => self.is_heavy = true,
+        set_shards(self, n: NumShards) => self.num_shards = Some(n),
+        set_validator_seats(self, n: NumSeats) => self.num_validator_seats = Some(n),
+        set_lightclient(self, n: usize) => self.num_lightclient = Some(n),
```

Comment:
Looking into testlib run_nodes implementation: https://github.com/near/nearcore/blob/92be6b559237530123851dcf829a3b9c7de5a281/test-utils/testlib/src/lib.rs#L63

I have two ideas:

* rename this method to set_lightclient_nodes_count
* remove this method and require users to pass “tracking_nodes_count” and “lightclient_nodes_count” in `NodeCluster:new` instead of “capacity”

Reply:
I'll go with the first. The second, I think, adds extra verbosity that's not that necessary.